lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -167,50 +171,61 @@ def store_column(column_title, format_string, new_data):
     """Checks calling sample_from_image on a loaded track reproduces the same result as is in the
     file."""
-    wrong_kymo_warning = (
-        "Photon counts do not match the photon counts found in the file. It is "
-        "possible that the loaded kymo or channel doesn't match the one used to "
-        "create this file."
-    )
     try:
-        if not np.allclose(
+        if np.allclose(
             track.sample_from_image((sampling_width - 1) // 2, correct_origin=True),
             track.photon_counts,
         ):
-            if np.allclose(
-                track.sample_from_image((sampling_width - 1) // 2, correct_origin=False),
-                track.photon_counts,
-            ):
-                return RuntimeWarning(
-                    "Photon counts do not match the photon counts found in the file. Prior to "
-                    "Pylake v1.1.0, the method `sample_from_image` had a bug that assumed the "
-                    "origin of a pixel to be at the edge rather than the center of the pixel. "
-                    "Consequently, the sampled window could be off by one pixel. This file was "
-                    "likely created using the incorrect origin. "
-                    "Note that Pylake loaded the counts found in the file as is, so if the "
-                    "used summing window was very small, there may be a bias in the counts."
-                    "To recreate these counts without bias invoke:"
-                    f"`track.sample_from_image({(sampling_width - 1) // 2}, correct_origin=True)`"
-                )
-            else:
-                return RuntimeWarning(wrong_kymo_warning)
+            return  # We're good
+
+        if np.allclose(
+            track.sample_from_image((sampling_width - 1) // 2, correct_origin=False),
+            track.photon_counts,
+        ):
+            return RuntimeWarning(
+                "Photon counts do not match the photon counts found in the file. Prior to "
+                "Pylake v1.1.0, the method `sample_from_image` had a bug that assumed the "
+                "origin of a pixel to be at the edge rather than the center of the pixel. "
+                "Consequently, the sampled window could be off by one pixel. This file was "
+                "likely created using the incorrect origin. "
+                "Note that Pylake loaded the counts found in the file as is, so if the "
+                "used summing window was very small, there may be a bias in the counts."
+                "To recreate these counts without bias invoke:"
+                f"`track.sample_from_image({(sampling_width - 1) // 2}, correct_origin=True)`"
+            )
+        else:
+            return RuntimeWarning(
+                "Photon counts do not match the photon counts found in the file. It is "
+                "possible that the loaded kymo or channel doesn't match the one used to "
+                "create this file."
+            )
```

Comment:
I see you already mention the processing in other comments. Also here it applies that another way to get a mismatch of kymo and tracks is when tracking was performed using `crop_by_distance`, but when loading the full kymo is given. Then the kymo matches the file but is not processed the same way.

Reply:
Ah yes, good point! Probably the most important place to mention it even as this might be the first thing users end up getting if they forget to do it. I changed it to:

```
                "Photon counts do not match the photon counts found in the file. It is "
                "possible that the loaded kymo or channel doesn't match the one used to "
                "create this file. Note that if you processed (e.g. sliced, cropped, flipped or "
                "downsampled) the kymograph prior to tracking, you will have to make sure that you "
                "crop the kymograph you supply to this function in the same way."
```

for the warning. The exception is the same except for one detail, namely that flipping cannot cause that indexing error and is omitted.